### PR TITLE
SUBMARINE 819. Create namespace for single user

### PR DESCRIPTION
**Pull request by Kenchu123**
### What is this PR for?

Create a namespace `submarine-user-test` of submarine for single user

### What type of PR is it?
[Feature]

### What is the Jira issue?

https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-819

### How should this be tested?

Create submarine:
```bash
cd submarine-cloud-v2
kubectl create ns submarine-operator-test
kubectl apply -f artifacts/example/crd.yaml
kubectl apply -f artifacts/examples/example-submarine.yaml -n submarine-operator-test
go builld -o submarine-operator
./submarine-operator
kubectl port-forward --address 0.0.0.0 service/traefik 32080:80 -n submarine-user-test
```

Delete submarine:
```bash
kubectl delete submarine example-submarine -n submarine-operator-test
```

### Screenshots (if appropriate)

https://user-images.githubusercontent.com/17617373/118001859-b1c7ad00-b379-11eb-8651-a829e90c050a.mov

### Questions:

* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

**Comment by ByronHsu** [comment]
The title should be "SUBMARINE-819. Create namespace for single user"

**Comment by Kenchu123** [comment]
> The title should be "[SUBMARINE-819](https://issues.apache.org/jira/browse/SUBMARINE-819). Create namespace for single user"

Thanks! I've changed it to the right one.

File: submarine-cloud-v2/controller.go [hunk 1]
```diff
@@ -1134,21 +1142,28 @@ func (c *Controller) syncHandler(workqueueItem WorkQueueItem) error {
...
+		if err != nil {
+			return err
+		}
+		klog.Info("Delete Namespace: ", newNamespace)
+
 		err = c.kubeclientset.CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
 		if err != nil {
 			return err
 		}
+		klog.Info("Delete Namespace: ", namespace)
```
Comment (kevin85421) [inline submarine-cloud-v2/controller.go:1155]: These lines can be removed.

File: submarine-cloud-v2/controller.go [hunk 2]
```diff
@@ -1134,21 +1142,28 @@ func (c *Controller) syncHandler(workqueueItem WorkQueueItem) error {
...
 
-		// Delete non-namespaced resources (ex: PersistentVolume)
```
Comment (kevin85421) [inline submarine-cloud-v2/controller.go:1142]: This line is very important. Do not remove it.

File: submarine-cloud-v2/controller.go [hunk 3]
```diff
@@ -1134,21 +1142,28 @@ func (c *Controller) syncHandler(workqueueItem WorkQueueItem) error {
...
 		if err != nil {
 			return err
 		}
 
-		err = c.kubeclientset.CoreV1().PersistentVolumes().Delete(context.TODO(), "submarine-tensorboard-pv--"+namespace, metav1.DeleteOptions{})
+		err = c.kubeclientset.CoreV1().PersistentVolumes().Delete(context.TODO(), "submarine-tensorboard-pv--"+newNamespace, metav1.DeleteOptions{})
 		if err != nil {
 			return err
 		}
+		klog.Info("Delete pv")
```
Comment (kevin85421) [inline submarine-cloud-v2/controller.go:1166]: This line can be removed.

File: submarine-cloud-v2/main.go [hunk 4]
```diff
@@ -19,17 +19,18 @@ package main
 
 import (
 	"flag"
-	kubeinformers "k8s.io/client-go/informers"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
-	"k8s.io/klog/v2"
```
Comment (kevin85421) [inline submarine-cloud-v2/main.go:26]: Why do we change these lines? My code formatter `gofmt` does not change these lines.

**Comment by Kenchu123** [comment]
@kevin85421 I've pushed new commits!

**Review by kevin85421 — APPROVED** [review]
LGTM

**Comment by xunliu** [comment]
@Kenchu123 @kevin85421 
The title of this PR is still `SUBMARINE 819. `, 
please check carefully next time before merging. 
Thanks.

**Comment by kevin85421** [comment]
@xunliu I am sorry for my negligence. I will be careful next time.

**Pull request closed without merging**